packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,510 @@
+  final T groupValue;
+
+  /// The callback that is called when a new option is tapped.
+  ///
+  /// This attribute must not be null.
+  ///
+  /// The segmented control passes the newly selected widget's associated key
+  /// to the callback but does not actually change state until the parent
+  /// widget rebuilds the segmented control with the new [groupValue].
+  ///
+  /// The callback provided to [onValueChanged] should update the state of
+  /// the parent [StatefulWidget] using the [State.setState] method, so that
+  /// the parent gets rebuilt; for example:
+  ///
+  /// ## Sample code
+  ///
+  /// ```dart
+  /// class SegmentedControlExample extends StatefulWidget {
+  ///   @override
+  ///   State createState() => new SegmentedControlExampleState();
+  /// }
+  ///
+  /// class SegmentedControlExampleState extends State<SegmentedControlExample> {
+  ///   final Map<int, Widget> children = const {
+  ///     0: const Text('Child 1'),
+  ///     1: const Text('Child 2'),
+  ///   };
+  ///
+  ///   int currentValue;
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return new Container(
+  ///       child: new SegmentedControl<int>(
+  ///         children: children,
+  ///         onValueChanged: (int newValue) {
+  ///           setState(() {
+  ///             currentValue = newValue;
+  ///           });
+  ///         },
+  ///         groupValue: currentValue,
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  final ValueChanged<T> onValueChanged;
+
+  @override
+  _SegmentedControlState<T> createState() => _SegmentedControlState<T>();
+}
+
+class _SegmentedControlState<T> extends State<SegmentedControl<T>> {
+  T _pressedKey;
+
+  void _onTapDown(T currentKey) {
+    setState(() {
+      _pressedKey = currentKey;
+    });
+  }
+
+  void _onTapUp(TapUpDetails event) {
+    setState(() {
+      _pressedKey = null;
+    });
+  }
+
+  void _onTapCancel() {
+    setState(() {
+      _pressedKey = null;
+    });
+  }
+
+  void _onTap(T currentKey) {
+    if (currentKey != widget.groupValue) {
+      widget.onValueChanged(currentKey);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> gestureChildren = <Widget>[];
+    int index = 0;
+    int selectedIndex;
+    int pressedIndex;
+    for (T currentKey in widget.children.keys) {
+      selectedIndex = (widget.groupValue == currentKey) ? index : selectedIndex;
+      pressedIndex = (_pressedKey == currentKey) ? index : pressedIndex;
+
+      final TextStyle textStyle = DefaultTextStyle.of(context).style.copyWith(
+        color: (widget.groupValue == currentKey) ?
+          CupertinoColors.white : CupertinoColors.activeBlue,
+      );
+      final IconThemeData iconTheme = new IconThemeData(
+        color: (widget.groupValue == currentKey) ?
+          CupertinoColors.white : CupertinoColors.activeBlue,
+      );
+
+      Widget child = widget.children[currentKey];
```

Comment:
Since we're adhering to the iOS design specs, we should center all the children so the example code you gave in the class doc looks right like in iOS.